geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureUtils.scala

```diff
@@ -48,17 +49,26 @@ object AvroSimpleFeatureUtils {
 
   def decodeAttributeName(s: String): String = attributeNameLookUp.getOrElseUpdate(s, decode(s))
 
-  def generateSchema(sft: SimpleFeatureType, withUserData: Boolean): Schema = {
+  def getNameEncoder(mangleNames: Boolean) =
+    if(mangleNames) encodeAttributeName _
+    else            (s: String) => StringUtils.replaceChars(s, "-", "_")
+
+  def generateSchema(sft: SimpleFeatureType,
+                     withUserData: Boolean,
+                     namespace: String = AVRO_NAMESPACE,
+                     mangleNames: Boolean = true): Schema = {
+    val nameEncoder = getNameEncoder(mangleNames)
+
     val initialAssembler: SchemaBuilder.FieldAssembler[Schema] =
-      SchemaBuilder.record(encodeAttributeName(sft.getTypeName))
-        .namespace(AVRO_NAMESPACE)
+      SchemaBuilder.record(nameEncoder(sft.getTypeName))
```

Comment:
Need to increment the version number of avro encoding since we are changing the schema i think